_extras/snippets_library/UCL_sge/14/depend_demo.snip

```diff
@@ -1,47 +1,59 @@
+which modules we currently have loaded in our environment. ([ANSYS](https://www.ansys.com/en-gb) is
+an engineering simulation product.)
 
 ```
-{{ site.host_prompt }} module load abinit
-{{ site.host_prompt }} module list
+{{ site.host_prompt }} module load ansys
 ```
 {: .bash}
-```
-Currently Loaded Modulefiles:
-  1) anaconda/python3                  5) intel-compilers-17/17.0.2.174     9) netcdf/4.4.1
-  2) mpt/2.16                          6) intel-cmkl-17/17.0.2.174         10) abinit/8.2.3-intel17-mpt214
-  3) intel-cc-17/17.0.2.174            7) gcc/6.2.0
-  4) intel-fc-17/17.0.2.174            8) fftw-3.3.5-intel-17.0.2-dxt2dzn
+ansys/2019.r3(73):ERROR:151: Module 'ansys/2019.r3' depends on one of the module(s) 'giflib/5.1.1'
+ansys/2019.r3(73):ERROR:102: Tcl command execution failed: prereq   giflib/5.1.1
 ```
 {: .output}
+This shows that the default `ansys` module will not run because it first needs `giflib/5.1.1` to be loaded.
+Some HPC systems will automatically load dependencies like this, but at the time of writing (June 2020) UCL's
+Myriad does not.
 
-So in this case, loading the `abinit` module
-also loaded a variety of other modules.
-Let's try unloading the `abinit` package.
+Let's load the `giflib` module:
+```
+{{ site.host_prompt }} module load giflib
+```
+{: .bash}
+```
+giflib/5.1.1(18):ERROR:151: Module 'giflib/5.1.1' depends on one of the module(s) 'gcc-libs/4.9.2'
+giflib/5.1.1(18):ERROR:102: Tcl command execution failed: prereq gcc-libs
+```
+{: .output}
+Here, we see that the `gflib` module itself also has a dependency, `gcc-libs`. So we have to load that
+first, then load `gflib`, and then finally load `ansys`.
 
 ```
-{{ site.host_prompt }} module unload abinit
-{{ site.host_prompt }} module list
+{{ site.host_prompt }} module load gcc-libs/4.9.2
+{{ site.host_prompt }} module load giflib/5.1.1
+{{ site.host_prompt }} module load ansys
 ```
 {: .bash}
 ```
-Currently Loaded Modulefiles:
-  1) anaconda/python3
+~/Scratch/.config is configured
+...
+...
+~/.mw doesn't exist - creating
 ```
 {: .output}
 
-So using `module unload` "un-loads" a module along with its dependencies.
-If we wanted to unload everything at once, we could run `module purge` (unloads everything).
+If you now use the `module list` command, you should see these three modules included in the list.
+
+To unload a specific module, e.g. `ansys`, run the command `module unload ansys`. (On some systems,
+this will also unload the modules it depends on. Currently this is not the case with Myriad.)
+
+If we wanted to unload everything at once (all modules), we could run `module purge` (unloads everything).
 
 ```
-{{ site.host_prompt }} module load abinit
 {{ site.host_prompt }} module purge
+{{ site.host_prompt }} module list
 ```
 {: .bash}
 ```
 No Modulefiles Currently Loaded.
 ```
 {: .output}
-
-Note that `module purge` has removed the `anaconda/python3` module as well as `abinit` and its dependencies.
```

Comment:
we should include also how to unload/purge the modules.

Reply:
I've already said that